### PR TITLE
Added task1 as pull request(correctlier)

### PR DESCRIPTION
**Pull request by IvanVuzh**

### CODE REVIEW

File: Csharp tasks/Task 1/Collection.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        {
+            order_collection.Add(order);
+        }
+
+        [Obsolete]
+        public void read_from_file(string filepath)
+        {
+            if (Validation.file_check(filepath))
+            {
+                JsonSchema schema = new JsonSchema();
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Collection.cs:28]: Use json serializer here as well.

File: Csharp tasks/Task 1/Collection.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                        //Console.WriteLine(new_order.GetType());
+                        if (new_order.all_check())
+                            order_collection.Add(new_order);
+                        else
+                            Console.WriteLine("Error adding new order to list");
+                    }
+                    else
+                        Console.WriteLine("Error in data for element №{0}", i+1);
+                }
+                //Console.WriteLine(json_for_order);
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Collection.cs:72]: remove comments

File: Csharp tasks/Task 1/Collection.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        public void new_order_input()
+        {
+            Order to_add = new Order();
+            to_add.console_input();
+            order_collection.Add(to_add);
+        }
+
+        public void write_to_file(string filepath)
+        {
+            //string res = "";
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Collection.cs:94]: remove comments

File: Csharp tasks/Task 1/Collection.cs [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+            {
+                order.print();
+            }
+        }
+
+        public void sort(string param)
+        {
+            if (order_collection.Count() != 0)
+            {
+                //order_collection.Sort( (x, y) =>
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Collection.cs:122]: remove comments

File: Csharp tasks/Task 1/Order.cs [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+                else
+                    order_status = "invalid";
+            }
+        }
+        public int Amount
+        {
+            get => amount;
+            set
+            {
+                if (Validation.check_id_or_amount(value))
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:50]: use ternary operator

File: Csharp tasks/Task 1/Order.cs [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+        protected string order_date;
+        protected string shipped_date;
+        protected string customer_email;
+
+        public int Id
+        {
+            get => id;
+            set 
+            {
+                if (Validation.check_id_or_amount(value))
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:28]: use ternary operator

File: Csharp tasks/Task 1/Order.cs [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+                else
+                    id = -1;
+            }
+        }
+        public string Order_status
+        {
+            get => order_status;
+            set
+            {
+                if (Validation.status_check(value))
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:39]: use ternary operator

File: Csharp tasks/Task 1/Order.cs [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+                else
+                    amount = -1;
+            }
+        }
+        public int Discount
+        {
+            get => discount;
+            set
+            {
+                if (Validation.check_discount(value))
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:61]: use ternary operator

File: Csharp tasks/Task 1/Order.cs [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+                else
+                    discount = -1;
+            }
+        }
+        public string Order_date
+        {
+            get => order_date;
+            set
+            {
+                if (Validation.date_check(value))
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:72]: use ternary operator

File: Csharp tasks/Task 1/Order.cs [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+                else
+                    order_date = "invalid";
+            }
+        }
+        public string Shipped_date
+        {
+            get => shipped_date;
+            set
+            {
+                if (Validation.date_check(value))
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:83]: use ternary operator

File: Csharp tasks/Task 1/Order.cs [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+        }
+
+        public string str_for_file()
+        {
+            string json_for_file = JsonSerializer.Serialize<Order>(this);
+            return json_for_file;
+        }      
+        public bool all_check()
+        {
+            bool res = true;
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:140]: unnecessary code (this method and public Order() as well)

File: Csharp tasks/Task 1/Order.cs [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
+            }
+            return res;
+        }
+
+        public void console_input()
+        {
+            string input;
+
+
+            Console.WriteLine("Enter id:");
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Order.cs:210]: remove copy paste

File: Csharp tasks/Task 1/Validation.cs [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+namespace PracticeTask1
+{
+    static class Validation
+    {
+        static public bool check_id_or_amount(object id)
+        {
+            if (check_if_int(id))
+            {
+                //Console.WriteLine("Id is int");
+                if (Convert.ToInt32(id) < 1)
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Validation.cs:18]: return !(Convert.ToInt32(id) < 1);

Comment (incora-dev) [reply Csharp tasks/Task 1/Validation.cs:18]: same for all methods

File: Csharp tasks/Task 1/Validation.cs [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+                //Console.WriteLine("Date of order must be made before shipping goods");
+                return false;
+            }
+            else
+                return true;
+        }
+
+        static public bool status_check(string order_status)
+        {
+            if (order_status == "paid" || order_status == "not paid" || order_status == "refunded")
```
Comment (incora-dev) [inline Csharp tasks/Task 1/Validation.cs:117]: make enum for order_status.

**Review by incora-dev — CHANGES_REQUESTED** [review]

**Review by tetianastoyko — APPROVED** [review]

**Pull request merged**